app/controllers/search_code.js

```diff
@@ -106,12 +107,15 @@ export default Ember.Controller.extend({
     },
 
     cancelSearch() {
+      let previousPath = this.get('previousPath');
       Ember.$("#searchText").blur();
       this.send("clearSearch", true);
       if(this.get("backToNewItem")) {
         this.transitionToRoute("items.new");
+      } else if (previousPath) {
+        this.transitionToRoute(previousPath);
```

Comment:
Can we only keep history.back()?
As it is more reliable and will also take to previous path and also less code in route to get the name of previous path

Reply:
I wasn't very sure about `history.back()`, that is why I wrote `previousPath` logic. Just to be on safer side I kept `history.back()` as a last option. If `history.back()` is more reliable than we can go for it. 👍

Reply:
Yes, I think it would better :)